src/redismodule.h

```diff
@@ -69,6 +69,20 @@
 #define REDISMODULE_HASH_CFIELDS    (1<<2)
 #define REDISMODULE_HASH_EXISTS     (1<<3)
 
+/* StreamID type. */
+typedef struct RedisModuleStreamID {
+    uint64_t ms;
+    uint64_t seq;
+} RedisModuleStreamID;
+
+/* Stream API flags. */
+#define REDISMODULE_STREAM_NONE 0
+#define REDISMODULE_STREAM_AUTOID (1<<0)
+#define REDISMODULE_STREAM_NOMKSTREAM (1<<1)
+#define REDISMODULE_STREAM_EXCLUSIVE (1<<2)
+#define REDISMODULE_STREAM_REVERSE (1<<3)
+#define REDISMODULE_STREAM_APPROX (1<<4)
```

Comment:
i feel it's a bit odd that we're mixing flags of different functions together (most will never be used in combination with each other).
maybe of these can maybe just be turned into a boolean argument (although that means that we can't extend it in the future)

Reply:
Just in case we want to allow some functions to share some flags, I made them use distinct bits. You use the macros anyway, right?

Reply:
Regarding boolean arguments vs flags: When reading code, which of the below is easier to understand?

```C
RedisModule_StreamIteratorStart(key, 1, start, end);
```

```C
RedisModule_StreamIteratorStart(key, REDISMODULE_STREAM_REVERSE, start, end);
```

Reply:
i agree it's makes it easier to read the code, and also opens the possibility for adding more flags in the future without modifying the API. 
but it's a bit odd that they share the same "bit" space.
i suppose i can live with it.

Reply:
I don't mind having a different bit space for each API function if you prefer that. Then maybe the name prefixes should reflect that too, e.g. `REDISMODULE_STREAM_ITERATOR_REVERSE`, etc.

Currently, each command has a check that only supported stream flags are used, exploiting the fact that all flags are distinct. If we make e.g. REVERSE = AUTOID, it's not possible for a check to forbid iteration with the AUTOID flag.

Reply:
yeah, if each of these flags applies only for one function, let's name them appropriately.